scripts/st2client-startup.sh

```diff
@@ -1,11 +1,14 @@
 #!/bin/bash
 
+counter=0
+
 # st2client startup and registration
-while true; do
+while [ "$counter" -lt 5 ]; do
```

Comment:
50 seconds might be too small here.

I'd prefer at least a couple of minutes of trying to register content before giving up.
If you think that's too many attempts, - I'm good if we go with the exponential backoff timeouts instead of every 5 sec.

Reply:
Well I think we should keep the number of checks down. The main reason for this is if there are authentication errors specifically to some kind of backend (LDAP, or others in the future) this could potentially lock or disable accounts. If you want more time to say like 120s (or more) then we could maybe add one more check with another exponent, or starting out at 10s instead of 5.

What are you thinking on the length of the timeout?

Reply:
I think starting with the initial `10s` as a timeout and multiplying it `* 1.5` on every iteration would look good enough.

And so eventually we'll get something like (~`2.5mins` total in `5` tries):
```
10s
15s
23s
34s
50s
```